### PR TITLE
Create and use getters for teams stuff

### PR DESCRIPTION
**Pull request by akalin-keybase**
Also combine members and usernames fields into one.

### CODE REVIEW

File: shared/teams/team/container.js [hunk 1]
```diff
@@ -31,13 +31,13 @@ const mapStateToProps = (state: TypedState, {routeProps, routeState}) => {
...
     _newTeamRequests: state.teams.getIn(['newTeamRequests'], I.List()),
-    numInvites: state.teams.getIn(['teamNameToInvites', teamname], I.Set()).size,
-    numRequests: state.teams.getIn(['teamNameToRequests', teamname], I.Set()).size,
-    numSubteams: state.teams.getIn(['teamNameToSubteams', teamname], I.Set()).size,
+    numInvites: Constants.getTeamInvites(state, teamname).size,
+    numRequests: Constants.getTeamRequests(state, teamname).size,
+    numSubteams: Constants.getTeamSubteams(state, teamname).size,
     loading: anyWaiting(state, Constants.teamWaitingKey(teamname)),
-    resetUserCount: state.teams.getIn(['teamNameToResetUsers', teamname], I.Set()).size,
+    resetUserCount: Constants.getTeamResetUsers(state, teamname),
```
Comment (buoyad) [inline shared/teams/team/container.js]: Needs a `.size`

File: shared/teams/team/members/container.js [hunk 2]
```diff
@@ -12,54 +12,55 @@ export type OwnProps = {
...
-      }
-      // b is active, should go later
-      return -1
-    } else if (!b.active) {
-      // b is inactive, should come first
-      return 1
-    }
-    if (listYouFirst && you) {
-      if (a.username === you) {
+  let returnArray = I.Set(memberInfo.values())
```
Comment (buoyad) [inline shared/teams/team/members/container.js]: :sheep: this is a little confusing, in other places we do `memberInfo.valueSeq().toArray()` (admittedly not  much clearer)

File: shared/teams/team/container.js (hunk 1 above)
Comment (akalin-keybase) [reply shared/teams/team/container.js]: D'oh! Not sure why flow didn't catch this. I tried adding some types to this file but it didn't seem to help. :(

File: shared/teams/team/members/container.js (hunk 2 above)
Comment (akalin-keybase) [reply shared/teams/team/members/container.js]: Done!

File: shared/teams/team/container.js (hunk 1 above)
Comment (chrisnojima) [reply shared/teams/team/container.js]: did you run flow coverage?

Comment (akalin-keybase) [reply shared/teams/team/container.js]: Thanks for reminding me, I'll do that

Comment (akalin-keybase) [reply shared/teams/team/container.js]: Done, was able to fix the typing (`ownProps` was unknown and was tainting everything else)

**Pull request merged**